named-tunnel-k8s/cloudflared.yaml

```diff
@@ -0,0 +1,89 @@
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: cloudflared
+spec:
+  selector:
+    matchLabels:
+      app: cloudflared
+  replicas: 2 # You could also consider elastic scaling for this deployment
+  template:
+    metadata:
+      labels:
+        app: cloudflared
+    spec:
+      containers:
+      - name: cloudflared
+        image: cloudflare/cloudflared:2021.4.0
+        args:
+        - tunnel
+        # Points cloudflared to the config file, which configures what
+        # cloudflared will actually do. This file is created by a ConfigMap
+        # below.
+        - --config
+        - /etc/cloudflared/config/config.yaml
+        - run
+        livenessProbe:
+          httpGet:
+            # Cloudflared has a /ready endpoint which returns 200 if and only if
+            # it has an active connection to the edge.
```

Comment:
However, being connected to the edge does not mean that this cloudflared is reachable by an eyeball.

In fact, we have no way of knowing that. Yet, I think it is an important detail for users. E.g., they may be tempted to do a rolling deployment by relying on k8s to probe this `/ready` endpoint, and that will cause downtime to eyeballs.

Reply:
At the very least we can add a `initialDelaySeconds` of 30sec or 60sec. It's not pretty, but it gives a much higher assurance that the tunnel is actually available from the edge perspective.

Reply:
QS lag is usually less than 10 seconds, I think that should be sufficient.